packages/core/src/dispatcher.ts

```diff
@@ -74,18 +74,24 @@ export default class Dispatcher extends EventEmitter {
    *
    * Logs error if subscribe fails.
    */
-  async _subscribe(): Promise<void> {
+  async _subscribe(force = false): Promise<void> {
     try {
-      await this._ipfs.pubsub.subscribe(
-        this.topic,
-        this.handleMessage,
-        // {timeout: IPFS_GET_TIMEOUT} // ipfs-core bug causes timeout option to throw https://github.com/ipfs/js-ipfs/issues/3472
-      )
-      this._log({peer: this._peerId, event: 'subscribed', topic: this.topic })
+      if (force || !(await this._ipfs.pubsub.ls()).includes(this.topic)) {
```

Comment:
Assumes 1 client for 1 node (otherwise this breaks). I think that’s a safe assumption as ipfs team says they weren’t designing for multi-client.

Reply:
Makes sense, thanks.